### ISSUE TITLE
[FR] Add per Object permissions and orgs

### ISSUE THREAD

**Issue by matmair** [inventree/InvenTree#2323]
**Problem**
The current permissions do not allow for multitenancy / multiple orgs in one instance.

**Suggested solution**
Add one of the libraries that allow per-object permissions.

**Describe alternatives you've considered**
I currently do this with a tenancy solution based on postgres schema-switching. Only works on postgres and could be broken / makes migrations difficult.

**Examples of other systems**
GeoNode has this for example.

**Do you want to develop this?**
Yes

**Comment by matmair** [comment]
https://django-guardian.readthedocs.io/en/stable/configuration.html

**Comment by SchrodingersGat** [comment]
@matmair what do you mean by "multiple orgs" in one instance here?

Do you mean supporting multiple InvenTree databases in one server instance, be it bare-metal or docker?

**Comment by matmair** [comment]
> @matmair what do you mean by "multiple orgs" in one instance here?
> 
> Do you mean supporting multiple InvenTree databases in one server instance, be it bare-metal or docker?

@SchrodingersGat different departments / sections / clubs in the entity that has an InvenTree instance.

One database but restricting access per part etc. to select groups / users.

**Comment by SchrodingersGat** [comment]
Ah ok, that is a nice idea, I can see it being quite difficult to implement?

**Comment by matmair** [comment]
> Ah ok, that is a nice idea, I can see it being quite difficult to implement?

Each permission check statement needs to be ammended but it should not be too bad. I will take a look after the current things are done.

**Comment by eeintech** [comment]
@matmair Do you intend to use the `django-guardian` library to implement this FR?

**Comment by matmair** [comment]
@eeintech yes, but it will take a while, it is fairly low on my priority list. There are a few devops, notification and security things I would like to see fixed before. And pricing needs a refactor.

**Comment by SchrodingersGat** [comment]
@matmair I think this is a lower priority than say the pricing refactor issues, shall we bump this down the road to (at least) `0.8.0`?

**Comment by matmair** [comment]
@SchrodingersGat I already started work on this, `0.8.0` or even a 0.7.x should be reasonable

**Comment by matmair** [comment]
See https://github.com/paperless-ngx/paperless-ngx/pull/2147/files
